perspective.el

```diff
@@ -524,24 +525,26 @@ See `persp-switch', `persp-get-quick'."
 (defun persp-next ()
   "Switch to next perspective (to the right)."
   (interactive)
-  (let* ((names (persp-names))
-         (pos (cl-position (persp-name (persp-curr)) names)))
-    (cond
-     ((null pos) (persp-find-some))
-     ((= pos (1- (length names)))
-      (if persp-switch-wrap (persp-switch (nth 0 names))))
-     (t (persp-switch (nth (1+ pos) names))))))
+  (when (persp-curr)
+    (let* ((names (persp-names))
+           (pos (cl-position (persp-name (persp-curr)) names)))
+      (cond
+       ((null pos) (persp-find-some))
+       ((= pos (1- (length names)))
+        (if persp-switch-wrap (persp-switch (nth 0 names))))
+       (t (persp-switch (nth (1+ pos) names)))))))
```

Comment:
This diffs MUCH better when you toggle the whitespace flag.

Seems like almost all of these could be refactored into a helper macro to make it a bit more clear...

I'm not sure about the `persp-last` one. You're mostly just verifying that there is _a_ perspective so it could probably be the same?

Reply:
```
(defmacro guard-perspecive (&rest body)
  (declare (indent 0))
  `(when (persp-curr)
     @,body))
```

(no clue what the name should be in)